pkg/api/validation/validation.go

```diff
@@ -628,6 +628,9 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource, fldPath *f
 	if len(hostPath.Path) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 	}
+
+	allErrs = append(allErrs, validatePathNoBacksteps(hostPath.Path, fldPath.Child("path"))...)
```

Comment:
@derekwaynecarr does the kubelet still use API validation to validate pods? what does it do if it is given a pod that validation determines is invalid? we don't want pods that include backsteps in their mount paths to run, but we don't want to kubelet to fall over if a pod like that existed previously and the kubelet gets upgraded.